src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs

```diff
@@ -78,6 +85,41 @@ internal virtual IList<string> FindPotentialSdkResolvers(string rootFolder, Elem
             return assembliesList;
         }
 
+        private DirectoryInfo[] GetSubfolders(string rootFolder, string additionalResolversFolder)
+        {
+            DirectoryInfo[] subfolders = null;
+            if (!string.IsNullOrEmpty(rootFolder) && FileUtilities.DirectoryExistsNoThrow(rootFolder))
+            {
+                subfolders = new DirectoryInfo(rootFolder).GetDirectories();
+            }
+
+            if (additionalResolversFolder != null)
+            {
+                var resolversDirInfo = new DirectoryInfo(additionalResolversFolder);
+                if (resolversDirInfo.Exists)
+                {
+                    var overrideFolders = resolversDirInfo.GetDirectories();
+                    var foldersSet = overrideFolders.ToHashSet(new DirInfoNameComparer());
+                    return overrideFolders.Concat(subfolders.Where(subfolder => foldersSet.Add(subfolder))).ToArray();
```

Comment:
You can simplify this with [HashSet.UnionWith](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.unionwith?view=netframework-4.7.2)